src/service/sdk_async_client.py

```diff
@@ -0,0 +1,82 @@
+"""
+A minimal async client for talking to KBase SDK generated servers (for example the Workspace).
+"""
+
+import aiohttp
+import random
+from typing import Any
+
+
+class ServerError(Exception):
+
+    def __init__(self, name=None, code=None, message=None, data=None, error=None):
+        # name and code are intentinally discarded
+        super().__init__(message)
+        self.message = '' if message is None else message
+        self.data = data or error or ''
+        # data = JSON RPC 2.0, error = 1.1
+
+    def __str__(self):
+        return f"{self.message}\n{self.data}"
+
+
+class SDKAsyncClient:
+    """
+    An async client for KBase SDK generated servers.
+    """
+
+    def __init__(self, url: str):
+        """
+        Create the client.
+
+        url - the url of the server.
+        """
+        self._session = aiohttp.ClientSession()
+        self._url = url
+
+    async def call(self, method: str, params: list[Any] = None, token: str = None):
+        """
+        Make a service call.
+
+        method - the name of the method, for example `Workspace.ver`.
+        params - the parameters for the method.
+        token - the user's KBase token, if any.
+
+        Returns will be as documented in the spec for the respective service.
+        """
+        body = {
+            'method': method,
+            'params': params or [],
+            'version': '1.1',
```

Comment:
making version a parameter and defaults to 1.1?

Reply:
A couple reasons I'm not sure about this:
* Why would the user ever care which version of the JSON-RPC protocol is used? It should be invisible to them either way
* The client is ported from the SDK `baseclient.py` and I have no idea if the code even works for JSON-RPC 2.0.